Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift

```diff
@@ -48,25 +48,22 @@ struct HTTPClientResponse {
 
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
 extension HTTPClientResponse.Body: AsyncSequence {
-    typealias Element = ByteBuffer
-    typealias AsyncIterator = Iterator
-
-    struct Iterator: AsyncIteratorProtocol {
-        typealias Element = ByteBuffer
+    public typealias Element = AsyncIterator.Element
 
+    public struct AsyncIterator: AsyncIteratorProtocol {
         private let stream: IteratorStream
 
         fileprivate init(stream: IteratorStream) {
             self.stream = stream
         }
 
-        func next() async throws -> ByteBuffer? {
+        public func next() async throws -> ByteBuffer? {
```

Comment:
Question rather than suggestion: should this be `mutating` to indicate that `next()` consumes from the stream?

Reply:
Makes sense to me. @fabianfett any objections?